modP2pImpl/src/org/aion/p2p/impl/comm/NodeMgr.java

```diff
@@ -239,17 +239,20 @@ public void timeoutCheck() {
     }
 
     private void timeoutOutBound() {
-        Iterator<Integer> outboundIt = getOutboundNodes().keySet().iterator();
-        while (outboundIt.hasNext()) {
-            int outBound = outboundIt.next();
-            INode node = getOutboundNodes().get(outBound);
-            if (System.currentTimeMillis() - node.getTimestamp()
-                > TIMEOUT_OUTBOUND_NODES) {
-                p2pMgr.closeSocket(
-                    node.getChannel(),
-                    "outbound-timeout node=" + node.getIdShort() + " ip=" + node.getIpStr());
-                outboundIt.remove();
+        try {
+            Iterator<Map.Entry<Integer, INode>> it = outboundNodes.entrySet().iterator();
+            while (it.hasNext()) {
+                Map.Entry<Integer, INode> entry = it.next();
+                if (System.currentTimeMillis() - entry.getValue().getTimestamp()
+                    > TIMEOUT_OUTBOUND_NODES) {
+                    p2pMgr.closeSocket(entry.getValue().getChannel(),
+                        "outbound-timeout ip=" + entry.getValue().getIpStr());
+                    it.remove();
+                }
             }
+        } catch (IllegalStateException e) {
+            e.printStackTrace();
+            p2pLOG.debug("timeoutOutbound IllegalStateException ", e.getMessage());
```

Comment:
`e.getMessage()` may return null causing another exception in the catch statement;
also `e.printStackTrace()` will not be made part of the logs

I suggest switching to using `info(String var1, Throwable var2)` from the `Logger` interface to avoid the above issues.

Reply:
https://github.com/aionnetwork/aion/pull/539/commits/828f0323156fc7c2993f709111d70b13f1cc4489